sources/osgShadow/ShadowMap.js

```diff
@@ -780,37 +781,61 @@ define( [
             var light = lightSource.getLight();
             var camera = this._cameraShadow;
 
+            var worldLightPos = this._worldLightPos;
+            var worldLightDir = this._worldLightDir;
+
             // make sure it's not modified outside our computations
             // camera matrix can be modified by cullvisitor afterwards...
+
             Matrix.copy( camera.getProjectionMatrix(), this._projectionMatrix );
             Matrix.copy( camera.getViewMatrix(), this._viewMatrix );
             var projection = this._projectionMatrix;
             var view = this._viewMatrix;
 
             // inject camera world matrix.
-            // from light current world/pos
+            // from light current world/pos and camera eye pos.
+            var lightMatrix = light.getPositionnedMatrix();
```

Comment:
I guess here we need to get LightSource information in the cull visitor. I guess it can be extracted from current renderstage positionned attribute. or looking into the stack node.